dask/dataframe/core.py

```diff
@@ -6789,8 +6788,10 @@ def _emulate(func, *args, udf=False, **kwargs):
     Apply a function using args / kwargs. If arguments contain dd.DataFrame /
     dd.Series, using internal cache (``_meta``) for calculation
     """
-    with raise_on_meta_error(funcname(func), udf=udf), check_numeric_only_deprecation():
-        return func(*_extract_meta(args, True), **_extract_meta(kwargs, True))
+    with check_numeric_only_deprecation():
+        with check_convert_dtype_deprecation():
```

Comment:
Can we pull this out and move it higher up in the callstack so it's closer to where `convert_dtype` is actually being specified? This seems too deep down in internals, lots of the `dask.dataframe` codebase uses `_emulate`